challenges/victor/databaseChallenge/migration.js

```diff
@@ -0,0 +1,112 @@
+// const mysql = require("mysql");
+
+// const connection = mysql.createConnection({
+//   host: "localhost",
+//   port: 3306,
+//   user: "root",
+//   password: "password",
+//   database: "headStormClientDB"
+// });
+
+// // assumes new database is created and being used in MySQL *****
+
+// connection.connect(function(err) {
+//   if (err) throw err;
+//   console.log("connected as id " + connection.threadId + "\n");
+//   dataMigrate(data);
+// });
+
+// grab old database data with a find() command
+data = {
+  recordID: 1234,
+  name: "Joe Smith",
+  cellPhone: "405.867.5309",
+  workPhone: "123.123.1234",
+  email: "joe_s@gmail.com",
+  address: "123 Vic Way, Dallas TX 75001",
+  basicWidget: 37,
+  advancedWidget: 12,
+  protectionPlan: true
+};
+
+function dataMigrate(data) {
+  // grab keys and values from JSON data object
+  const keys = Object.keys(data);
+  const {
+    recordID,
+    name,
+    cellPhone,
+    workPhone,
+    email,
+    address,
+    basicWidget,
+    advancedWidget,
+    protectionPlan
+  } = data;
+
+  // create new mySQL table with JSON returned from old database
+  console.log(
+    "CREATE TABLE records (" +
```

Comment:
What would be the pros/cons of saving this in a single table (records) vs normalizing the data into specific entities (e.g. Customer, Order, etc)?

Reply:
I thought the goal was to have matching data from the noSQL database in the SQL database. That strategy would allow you to compare the data in both databases easily as they would be identical structurally. However thinking about it more, If I were to break the data into customer and order tables it would allow for easier querying of the database for instance checking how many orders a specific customer has ordered. Also it would help prevent duplicate data in the database for the multiple orders by the same customer.